superset-frontend/src/dashboard/components/gridComponents/Tabs.jsx

```diff
@@ -256,10 +264,51 @@ export class Tabs extends React.PureComponent {
     }
   };
 
+  // the initial color map is generated on save and catches all rendered charts
```

Comment:
Currently the initial color map is stored in the dashboard metadata at save but fails to map colors of charts that are hidden within tabs as those are not rendered. This checks for new colors popping in the color map that are not in the stored metadata and updates the metadata accordingly. The stored color map is then used by the viz in Explore to respect the colors of the dashboard context.

Reply:
Ideally this would not be necessary. A chart could expose a post-render hook that will individually check whether its color map is correctly updated in the stored metadata. However, this might have performance implications as multiple requests would need to be made. A larger refactor of the implementation would be required to account for efficiency in updating the color map and performance.